sql/2020/11_Security/version-evolution-top-technologies.sql

```diff
@@ -0,0 +1,50 @@
+#standardSQL
+# Distribution of the different version of the top 20 technologies used on the web.
+
+SELECT
+  category,
+  app,
+  info,
+  month,
+  client,
+  pct
+FROM (
+  SELECT
+    info,
+    tech.category AS category,
+    tech.app AS app,
+    LEFT(_TABLE_SUFFIX, 10) AS month,
+    IF(ENDS_WITH(_TABLE_SUFFIX, '_desktop'), 'desktop', 'mobile') AS client,
+    COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY _TABLE_SUFFIX, tech.category, tech.app) AS pct
+  FROM
+    `httparchive.technologies.*` AS tech
+  INNER JOIN (
+    SELECT
+      category,
+      app,
+      COUNT(0) AS num
+    FROM
+      `httparchive.technologies.*`
+    WHERE
+      REGEXP_CONTAINS(_TABLE_SUFFIX, r'^20(20|19).*')
+    GROUP BY
+      category,
+      app
+    ORDER BY num DESC
+    LIMIT 20
```

Comment:
This will mean you're limiting it to the top technologies for 2019 and 2020 rather than the top technologies as of now (or Aug 2020). That may be fine, but thought I'd point it out just to make sure you get that.

Reply:
That's as intended; since we're looking at the software over the period of a year, I thought it made sense to look at those that have been most popular during this period.